src/commands/Invoke.ts

```diff
@@ -1,76 +1,122 @@
         Object.keys(payload).forEach((key) => {
             form.append(key, payload[key]);
         })
-        body = form;
-    } else {
-        headers["Content-Type"] = "application/x-www-form-urlencoded";
-        body = queryString.stringify(flatten(payload), { encode:false });
+        this.init.body = form
+        return this
     }
-    return new Request({
-        href: `${process.env.PAXFUL_DATA_HOST}${url}`
-    }, {
-        method: "POST",
-        headers,
-        agent: config.proxyAgent,
-        body
-    });
-}
 
-const convertResponse = (response: Response) => {
-    const contentType = response.headers.get("Content-Type");
-    if (contentType) {
-        if( contentType.startsWith("text/") ) {
-            return response.text()
-        } else if (contentType.indexOf("json") > 0) {
-            return response.json()
-        }
+    public withJsonData(data?: AnyJson): RequestBuilder {
+        this.withHeader("Content-Type", "application/json")
+        this.init.body = JSON.stringify(data || {})
+        return this
+    }
+
+    public withAuthorization(credentials: Credentials): RequestBuilder {
+        this.withHeader("Authorization", `Bearer ${credentials.accessToken}`)
+        return this
+    }
+
+    public acceptJson(): RequestBuilder {
+        this.responseParser = async response => await response.json()
+        this.withHeader("Accept", `application/json`)
+        return this
+    }
+
+    public acceptText(): RequestBuilder {
+        this.responseParser = async response => await response.text()
+        return this
     }
-    return response.blob()
-        .catch(() => (
-            response.arrayBuffer()
+
+    public acceptBinary(): RequestBuilder {
+        this.responseParser = async response => {
+            return response.blob()
                 .catch(() => (
-                    response.buffer()
+                    response.arrayBuffer()
+                        .catch(() => (
+                            response.buffer()
+                        ))
                 ))
-        ))
+        }
+        return this
+    }
+
+    public build(): [Request, ResponseParser]  {
+        return [new Request(this.url, this.init), this.responseParser]
+    }
 }
 
 /**
+ * Executes request with the user credentials applied.
  * Retrieves personal access token and refresh token.
  *
- * @param url
- * @param payload
+ * @param requestBuilder
  * @param credentialStorage
  * @param config
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
-export default async function invoke(url: string, config: ApiConfiguration, credentialStorage?: CredentialStorage, payload?: InvokeBody): Promise<any> {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default async function executeRequestAuthorized(requestBuilder: RequestBuilder, config: ApiConfiguration, credentialStorage?: CredentialStorage): Promise<any> {
```

Comment:
Should you not change here? `Promise<any>`

Reply:
On some reason, typescript complaining here in case of AnyPromise is set there.